### PR TITLE
zathura: 0.5.6 -> 0.5.8, all plugins updated

### LINKED ISSUE #334368: Update request: zathura-pdf-mupdf 0.4.2 → 0.4.5
- Package name: zathura-pdf-mupdf
- Latest released version: 0.4.5

- Current version on the unstable channel: 0.4.2
- Current version on the stable/release channel: 0.4.1

- [ x ] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

This would be nice because as https://github.com/pwmt/zathura/issues/601 states
this would fix zathura segfaulting when zooming to fast.

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by smancill**
## Description of changes

Update zathura and plugins.

- zathura-cb: 0.1.10 -> 0.1.11
- zathura-djvu: 0.2.9 -> 0.2.10
- zathura-pdf-mupdf: 0.4.2 -> 0.4.4 (waiting for mupdf 1.24 to land on `master` #332385)
- zathura-pdf-poppler: 0.3.2 -> 0.3.3
- zathura-ps: 0.2.7 -> 0.2.8

For the plugins, their `data/meson.build` now requires `desktop-file-utils` (for `desktop-file-validate`) and `appstream-glib` (for `appstream-util`). Or the alternative is to patch the `required: get_option('tests')` argument for `find_program`.

Supersedes #325136.

Closes #334368 (last release is 0.4.4)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by smancill** [comment]
Making it a draft for now until mupdf 1.24 is available. Or should I go to `staging`?

File: pkgs/applications/misc/zathura/core/default.nix [hunk 1]
```diff
@@ -28,15 +29,24 @@
...
 
   src = fetchFromGitHub {
     owner = "pwmt";
     repo = "zathura";
     rev = finalAttrs.version;
-    hash = "sha256-lTEBIZ3lkzjJ+L1qecrcL8iseo8AvSIo3Wh65/ikwac=";
+    hash = "sha256-k6DEJpUA3s0mGxE38aYnX7uea98LrzevJhWW1abHo/c=";
   };
 
+  patches = lib.optionals stdenv.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/zathura/core/default.nix:41]:
```suggestion
  patches = [
```

File: pkgs/applications/misc/zathura/cb/default.nix [hunk 2]
```diff
@@ -9,22 +9,26 @@
...
+    hash = "sha256-TiAepUzcIKkyWMQ1VvY4lEGvmXQN59ymyh/1JBcvvUc=";
   };
 
   nativeBuildInputs = [
     meson
     ninja
     pkg-config
     gettext
+    desktop-file-utils
+    appstream-glib
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/zathura/cb/default.nix:31]: appstream-glib goes into buildInputs

Comment (smancill) [reply pkgs/applications/misc/zathura/cb/default.nix:31]: It's only used by a test.

**Comment by smancill** [comment]
@ofborg build zathura

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 337383` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>vimPlugins.openscad-nvim</li>
    <li>zathura</li>
  </ul>
</details>

**Review by SuperSandro2000 — DISMISSED** [review]
fails to build

```
Dependency mupdf found: NO. Found 1.23.6 but need: '>=1.24'
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency mupdf found: NO (tried pkgconfig and cmake)
Has header "mupdf/fitz/version.h" : YES 
Has header "mupdf/fitz.h" : YES 
Has header "mupdf/pdf.h" : YES 
Library mupdf found: YES
meson.build:45:4: ERROR: Problem encountered: mupdf 1.24 or newer is required
A full log can be found at /build/zathura-pdf-mupdf-0.4.4/build/meson-logs/meson-log.txt
error: builder for '/nix/store/bvc3qjc4jrf7lndwlr0g00yw8wsscffv-zathura-pdf-mupdf-0.4.4.drv' failed with exit code 1;
```

Also it would be better to do each update in its own commit. Then ofborg would also build all packages.

**Comment by smancill** [comment]
It works, but it just needed a rebase. The nixpkgs-review does the merge so that's why it worked too.

**Review by fpletz — APPROVED** [review]

**Pull request merged**